src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -58,5 +59,41 @@ public void AddOrUpdatePackageSourceMappingSourceItem(PackageSourceMappingSource
 
             _settings.SaveToDisk();
         }
+
+        public void SavePackageSourceMappings(IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems)
+        {
+            if (packageSourceMappingsSourceItems == null)
+            {
+                throw new ArgumentNullException(nameof(packageSourceMappingsSourceItems));
+            }
+
+
+            var existingSettingsLookup = GetPackageSourceMappingItems();
+
+            foreach (var sourceMappingItem in packageSourceMappingsSourceItems)
+            {
+                if (existingSettingsLookup != null)
+                {
+                    AddOrUpdatePackageSourceMappingSourceItem(sourceMappingItem);
+                }
+            }
+
+            //Remove all old mappings not in new mappings
+            if (existingSettingsLookup != null)
```

Comment:
I believe you may need to reload this value. 
Because `existingSettingsLookup` is stale now, so you may not see new values here.

Reply:
Yeah I won't see any new values, but I don't think I need to see the mappings that were just added/updated. I really only need to see the ones that are in the config currently but aren't the current list of mappings displayed in the UI. These will still be in existingSettingsLookup even if I add new ones